lib/galaxy/datatypes/binary.py

```diff
@@ -236,6 +236,49 @@ def sniff(self, filename):
         return False
 
 
+class Bref3(Binary):
+    """Bref3 format is a binary format for storing phased, non-missing genotypes for a list of samples."""
+
+    file_ext = "bref3"
+
+    def __init__(self, **kwd):
+        super().__init__(**kwd)
+        self._magic = binascii.unhexlify("7a8874f400156272")
+
+    def sniff(self, filename):
+        """
+        Try to guess if the file is a bref3 file.
+        >>> from galaxy.datatypes.sniff import get_test_fname
+        >>> fname = get_test_fname('affy_v_agcc.cel')
+        >>> Bref3().sniff(fname)
+        False
+        >>> fname = get_test_fname('ref.bref3')
+        >>> Bref3().sniff(fname)
+        True
+        """
+        try:
+            header = open(filename, 'rb').read(8)
+            if header == self._magic:
+                return True
+            return False
+        except Exception:
+            return False
```

Comment:
```suggestion
    def sniff_prefix(self, sniff_prefix):
        return sniff_prefix.startswith_bytes(self._magic)
```

that should do the same without repeatedly opening the file.